pkg/flow/flow.go

```diff
@@ -210,7 +207,7 @@ func (c *Flow) Run(ctx context.Context) {
 				}
 
 				level.Debug(c.log).Log("msg", "handling component with updated state", "node_id", updated.NodeID())
-				c.loader.EvaluateDependencies(nil, updated)
```

Comment:
There was a lot of passing around a nil `*vm.Scope` which I removed. If I was missing some context and need to put it back let me know.